Committed for 2.34

commit 36231bee7ab36d59dd121ea85b91411ae86945f3
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Mon Feb 1 19:39:48 2021 +0100

    hurd TIOCFLUSH: Cope BSD 4.1 semantic
    
    BSD 4.1 did not have an argument for TIOCFLUSH, BSD 4.2 added it. There
    are still a lot of applications out there that pass a NULL argument to
    TIOCFLUSH, so we should rather cope with it.

--- a/sysdeps/mach/hurd/ioctl.c
+++ b/sysdeps/mach/hurd/ioctl.c
@@ -112,6 +112,10 @@
 	  /* We don't want to advance ARG since it will be used to copy out
 	     too if IOC_OUT is also set.  */
 	  void *argptr = arg;
+	  int zero = 0;
+
+	  if (request == TIOCFLUSH && !argptr)
+	    argptr = &zero;
 
 	  /* Pack an argument into the message buffer.  */
 	  void in (unsigned int count, enum __ioctl_datum type)
